fedimint-cli/src/lib.rs

```diff
@@ -516,13 +531,16 @@ impl FedimintCli {
                 hash: env!("FEDIMINT_BUILD_CODE_VERSION").to_string(),
             }),
             Command::Client(ClientCmd::Restore { secret }) => {
-                let client = cli
+                let mut client_builder = cli
                     .build_client_builder(&self.module_inits, None)
                     .await
-                    .map_err_cli_msg(CliErrorKind::GeneralFailure, "failure")?
-                    .build_restoring_from_backup(ClientSecret::<PlainRootSecretStrategy>::new(
-                        secret,
-                    ))
+                    .map_err_cli_general()?;
+                client_builder
```

Comment:
`devimint` backup-and-restore test fails because `wipe` command only deletes module state, not the secret. So here we have a few options:
- do what we're doing everywhere else, that is, try loading it first and write it only if absent
- load it, expect it to be there, and compare it with the provided secret, and proceed only if the two match
- `wipe` could also be made to delete the global client state

We might also not need `wipe` now that we have devimint and can just fork clients?

So for now we just go with:
- load secret
- if it exists, compare, and return err if different
- if it doesn't exist, store it
- continue with restoration

Reply:
Done